github_server/action.yml

```
@@ -1,53 +1,143 @@
+          echo "MAX_SECONDS_FOR_DOCKER=${{ inputs.MAX_SECONDS_FOR_DOCKER }}" >> $GITHUB_ENV
+          echo "DA_ADMIN_EMAIL=admin@example.com" >> $GITHUB_ENV
+          # Login will fail if this is "password"
+          echo "DA_ADMIN_PASSWORD=@123abcdefg" >> $GITHUB_ENV
         shell: bash
-      - name: Download and start docker
+      - id: api_key_output_step
+        run: |
+          echo "DOCASSEMBLE_DEVELOPER_API_KEY=$DA_ADMIN_API_KEY" >> "$GITHUB_OUTPUT"
+          echo "DA_ADMIN_EMAIL=$DA_ADMIN_EMAIL" >> "$GITHUB_OUTPUT"
+          echo "DA_ADMIN_PASSWORD=$DA_ADMIN_PASSWORD" >> "$GITHUB_OUTPUT"
+        shell: bash
+
+      - name: "ALKiln github_server: Download and start docker"
         run: |
           mkdir /tmp/config
-          echo "$CONFIG_CONTENTS" > /tmp/config/myconfig.yml 
-          docker pull brycewilley/smoldocassemble
-          docker run --env DA_ADMIN_EMAIL="admin@example.com" --env DA_ADMIN_PASSWORD="password" --env DA_ADMIN_API_KEY="${{ env.DA_ADMIN_API_KEY }}" --env DA_CONFIG=/tmp/config/myconfig.yml --volume /tmp/config:/tmp/config --cap-add SYS_PTRACE --memory="4gb" -d -p 80:80 jhpyle/docassemble
-          # TODO before merge: do a while loop here until `localhost` shows "It's ready"
-          sleep 480
-          curl localhost
-        shell: bash
-      - name: Run Kiln
-        # TODO: Research how to get this to always be the same branch we're on
-        uses: SuffolkLITLab/ALKiln@v5
+          echo "${{ inputs.CONFIG_CONTENTS }}" > /tmp/config/myconfig.yml
+          docker pull jhpyle/docassemble
+          docker run --env DA_ADMIN_EMAIL="${{ env.DA_ADMIN_EMAIL }}" --env DA_ADMIN_PASSWORD="${{ env.DA_ADMIN_PASSWORD }}" --env DA_ADMIN_API_KEY="${{ env.DA_ADMIN_API_KEY }}" --env DA_CONFIG=/tmp/config/myconfig.yml --volume /tmp/config:/tmp/config --cap-add SYS_PTRACE --memory="4gb" -d -p 80:80 jhpyle/docassemble
+        shell: bash
+      - name: "ALKiln github_server: Wait for server to start"
+        run: |
+          # ALKiln github_server: Wait for server to start
+
+          # Reset the SECONDS variable
+          SECONDS=0
+          # Wait 1 minute before pinging the server constantly
+          sleep 60
+
+          # Use a variable instead of checking $response so we can control to avoid race
+          # conditions from preventing the success log message.
+          ready="false"
+
+          while [[ $ready == "false" ]]; do
+            echo "Time ellapsed: $SECONDS seconds"
+
+            # Check if max seconds till loading has passed
+            if [[ $SECONDS -ge $MAX_SECONDS_FOR_DOCKER ]]; then
+              echo "ALKiln GitHub Server Error: Timed out waiting for docker container to serve the site. If you want to give your docker container more timem to install, set the MAX_SECONDS_FOR_DOCKER input for this action."
+              # Exit with failure, or allow this to keep going?
+              break
```

IMO exiting here is a good idea.

```suggestion
              exit 1
```